examples/gsd/_03_dmo_evaluation_on_wb_level.py

```diff
@@ -0,0 +1,251 @@
+    )
+)
+print(gs_tp_fp_fn)
+
+
+# %%
+# Based on the positive matches,
+# we can now extract the DMO data from detected and reference data that is to be compared.
+# To combine this data, we use the :func:`~mobgap.gsd.evaluation.get_matching_gs` function.
+
+from mobgap.gsd.evaluation import get_matching_gs
+
+gs_matches = get_matching_gs(metrics_detected=detected_dmo, metrics_reference=reference_dmo, matches=gs_tp_fp_fn)
+print(gs_matches)
+
+# %%
+# Estimate Errors in DMO data
+# ----------------------------
+# The DMO data can now be compared gait sequence by gait sequence.
+# To estimate discrepancies between the detected and reference DMOs,
+# the :func:`~mobgap.gsd.evaluation.apply_transformations` can be used.
+# As input, it receives the matching DMO data and a list of transformations that should be applied to the data.
+# A transformation is characterized as a function that takes some subset of the input dataframe,
+# performs some operation on it, and returns a series as output.
+# The transformations are a list of tuples containing the DMO of interest as the first element and the error functions
+# applied to the detected and reference values as the second element.
+# This way, you can also define custom error functions and pass them as transformations.
+# Note that the columns of the detected and reference values are expected to be named `detected` and `reference`
+# per default.
+# For the standard error metrics (error, absolute error, relative error, absolute relative),
+# the :func:`~mobgap.gsd.evaluation.get_default_error_metrics` returns the correct transformations.
+from mobgap.gsd.evaluation import apply_transformations, get_default_error_metrics
+
+gs_errors = apply_transformations(gs_matches, get_default_error_metrics())
+print(gs_errors)
+
+# %%
+# Modify Transformation Functions
+# -------------------------------
+# For some DMOs, it might occur that the reference value is 0, which would lead to a division by zero when calculating
+# the relative error. In this example, this happens for the `n_turns` parameter.
+# Per default, the function raises a warning when zero division occurs and sets the relative error to NaN.
+# To suppress the warning, the default error function arguments can be modified by defining adapted functions
+# setting the `zero_division_hint` parameter to `np.nan`.
+# We set the name of the adapted functions to the names of the original functions,
+# as the columns in the output dataframe are named after the functions.
+# The transformations list then needs to be updated with the adapted functions.
+# This way, no warning will be raised when zero division occurs.
+# In the same manner, the default names `detected` and `reference` of the two input columns for the error functions
+# can be overwritten by custom names.
+from mobgap.gsd.evaluation import abs_error, abs_rel_error, error, rel_error
+
+rel_err_suppressed_warning = lambda x: rel_error(x, zero_division_hint=np.nan)
```

Comment:
We should think about this general usecase -> Is there a more elegant way to handle custom names?
In this case here, it might be easier to just define a custom function instead of a lambda

Reply:
What we could do, is allow for a tuple instead of just a callable? So `(name, callable)`. This might also solve issues with other callables that don't have names